File: DBS_10352505_CA__Python/calculator.py
import math


class Calculator:

    def __init__(self):
        self.__input = '0'
        self.__operation = ''

    def add(self, first, second):
        return first + second

    def multiply(self, first, second):
        return first * second

    def subtract(self, first, second):
        return first - second
    
    def divide(self, first, second):
        return first / second
        
    def percentage(self, first, percent_change):
        return first + ( (first /100) * percent_change )

    def square_root(self, first):
        if first < 0:
            return 0
        else:
            return  math.sqrt(first) 

    def square(self, first):
        return first * first
    
    def exponentiate(self, baseNum, exponentNum):
        return math.pow(baseNum, exponentNum)
    
    def cube(self, baseNum):
        return math.pow(baseNum, 3)
    
    def log10(self, first):
        if first < 0:
            return 0
        else:
            return math.log10(first)
        #math.log10(x) - Return the base-10 logarithm of x. 
        #This is usually more accurate than log(x, 10).

        
        
        
    

    def input(self, value):
        if value == '#':
            self.__input = '0'
        elif value == '+':
            self.__operation = value
        elif self.__operation != '':
            self.__input = str(self.add(int(self.__input), int(value)))
        else:
            self.__input += str(value)
        #print self.__input

    def getOutput(self):
        if self.__input == '0E':
            return 0
        return int(self.__input)

# --------------------------------------------------------------------------
    
    def screenDisplay(self):
        print '\n' * 50	
        print "\n\n\n\n\n"
        print ("******************************************")
        print ("************** Calculator ****************")
        print ("******************************************\n\n")
    
        print ("1)  Addition")
        print ("2)  Subtraction")
        print ("3)  Multiplication")
        print ("4)  Division")
        print ("5)  Percentage")
        print ("6)  Square Root")
        print ("7)  Square")
        print ("8)  Exponentiate")
        print ("9)  Cube")
        print ("10) Log10\n")
        print ("E to exit.\n")
 
    
    def getUserNumber(self):
        isANumber = False
        
        while (isANumber == False):
            userInput = raw_input()                
            try:
                number = float(userInput)
                print ("You entered %s"  %number)
                return number
                isANumber = True
            except:
                print("%s is not a valid number - please enter again: " %userInput)        
    
    
 

File: TDT_Calculator/TDT_Calculator/Calculator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TDT_Calculator
{
    public class Calculator
    {

        bool IsTesting = false;
        Object[] Values;
        int CurrentIndex = 0;

        public int GetCurrentIndex()
        {
            return CurrentIndex;
        }

        public void SetTestValues(Object[] values)
        {
            IsTesting = true;
            Values = values;
        }

        // 1) AddNums Method - add 2 numbers and returns the result
        public double AddNums(double first, double second)
        {
            return first + second;
        } // end AddNums method


        // 2) SubTractNumbers Method - subtracts 2nd number from 1st number
        public double SubtractNumbers(double first, double second)
        {
            return first - second;
        } // end SubtractNumbers method


        // 3) Multiply_Numbers Method - subtracts 2nd number from 1st number
        public double MultiplyNumbers(double first, double second)
        {
            return first * second;
        } // end MultiplyNumbers method



        // 4) Divide_Numbers Method - subtracts 2nd number from 1st number
        public double DivideNumbers(double first, double second)
        {
            return first / second;
        } // end DivideNumbers method




        // 5) Percentages Button
        public double Percentage_Change(double first, double percent_change)
        {
            return first + ((first / 100) * percent_change);
        }


        // 6) Square Root
        public double Square_Root(double first)
        {
            return Math.Sqrt(first);
        }


        // 7) Square a number
        public double Square(double first)
        {
            return first * first;
        }


        // 8) Exponentiate
        public double Exponentiate(double baseNum, double exponentNum)
        {
            return Math.Pow(baseNum, exponentNum);
        }

        // 9) Cube
        public double Cube(double first)
        {
            return Math.Pow(first, 3);
        }


        // 10) Log10
        public double Log10(double first)
        {
            return Math.Log10(first);
        }






        // method to get a valid number from the user.
        public double GetUserNumber(string userInput)
        {
            bool isANumber = false;            
            double userNumber = 0;

            while (!isANumber)
            {
                isANumber = double.TryParse(userInput, out userNumber);
                if (!isANumber)
                {
                    userInput = GetNextValue();
                }
            }
            return userNumber;
        }

        public string GetNextValue()
        {
            if (!IsTesting)
            {
                Console.Clear();
                Console.WriteLine("Please enter a valid number: ");
                return Console.ReadLine();
            }
            return Values[CurrentIndex++].ToString();
        }

        public void ScreenDisplay()
        {
            Console.Clear();
            Console.WriteLine("******************************************");
            Console.WriteLine("***************Calculator*****************");
            Console.WriteLine("******************************************\n\n");

            Console.WriteLine("1)  Addition");
            Console.WriteLine("2)  Subtraction");
            Console.WriteLine("3)  Multiplication");
            Console.WriteLine("4)  Division");
            Console.WriteLine("5)  Percentage");
            Console.WriteLine("6)  Square Root");
            Console.WriteLine("7)  Square");
            Console.WriteLine("8)  Exponentiate");
            Console.WriteLine("9)  Cube");
            Console.WriteLine("10) Log10\n");
            Console.WriteLine("E to exit.\n\n");
        }








    } //end class MyCalculator
} // end Namespace
